src/OSPSuite.Infrastructure.Export/DataRepositoryExportTask.cs

```diff
@@ -108,8 +108,15 @@ private static string tableNameFrom(DataColumn baseGrid)
          var tableName = baseGrid.Repository != null ? baseGrid.Repository.Name : defaultTableName;
          if (string.IsNullOrEmpty(tableName))
             tableName = defaultTableName;
+         var correctedTableName = new string(FileHelper.RemoveIllegalCharactersFrom(tableName).Take(Constants.MAX_NUMBER_OF_CHAR_IN_TABLE_NAME).ToArray());
```

Comment:
2 remarks:
* `[` and `]` are two chars which are not allowed in the excel sheet names but are allowed in the file names, so I guess `FileHelper.RemoveIllegalCharacters` will not remove them
* Why is `.ToArray()` required?

Reply:
To be honest this line was the same before: I have just created the correctedTableName variable to remove the ' before returning it. The way it is written, it needs to be an array for new string() to work. Do you think this line should be somehow refactored?

About the [ and ] characters: I added functionality removing them (also a test for that).

Reply:
>Do you think this line should be somehow refactored?

Just checked, that Take returns IEnumerable<char> and not a string, that's why. Is ok then.